src/components/Pagination/__tests__/Pagination.spec.js

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import toJson from 'enzyme-to-json';
+import Pagination from '../Pagination';
+
+describe('<Pagination> that renders a pagination component', () => {
+    const CURRENT_PAGE = 6;
+    const onClick = jest.fn();
+    let wrapper;
+    beforeEach(() => {
+        wrapper = shallow(
+            <Pagination
+                currentPage={CURRENT_PAGE}
+                maxPageButtons={8}
+                totalPages={20}
+                onClick={onClick}
+                prevLabel="Previous"
+                nextLabel="Next"
+                firstLabel="First"
+                lastLabel="Last"
+            />
+        );
+    });
+
+    afterEach(() => {
+        jest.resetAllMocks();
+    });
+
+    it('should render correctly with default props', () => {
+        wrapper = shallow(<Pagination totalPages={20} />);
+
+        expect(toJson(wrapper)).toMatchSnapshot();
+    });
+    it('should render correctly with all props', () => {
```

Comment:
Can we add test coverage for all funky corner cases:
* `maxPageButtons >= totalPages`
* `currentPage = 1`
* `currentPage = totalPages` (currently broken)
* gaps between `start ... current block ... end`
* `maxPageButtons = 1, currentPage != 1` (currently broken)

They don't have to be snapshots, but something like `given props {....} we expect to see buttons 1, 4, 5, 6 and 1 hellip`

Reply:
added some